File: Warehousing.UI/src/app/admin/models/role.ts
import { RoleCategory } from "./RoleCategory";
import { RolePermission, RolePermissionDto } from "./rolePermission";
import { RoleProduct } from "./RoleProduct";
import { RoleSubCategory } from "./RoleSubCategory";

export interface Role {
    id: number;
    code: string;
    nameEn: string;
    nameAr: string;
    rolePermissions: RolePermission[];
    roleCategories : RoleCategory[],
    roleProducts : RoleProduct[],
    roleSubCategories : RoleSubCategory[]
}
export interface RoleDto {
    id: number;
    code: string;
    nameEn: string;
    nameAr: string;
    permissions: RolePermissionDto[];
    categoryIds : number[],
    productIds : number[],
    subCategoryIds : number[],
    printerConfigurationId?: number | null
}

export interface RoleDtoForAdd {
    id: number;
    code: string;
    nameEn: string;
    nameAr: string;
    rolePermissionIds: number[];
    categoryIds : number[],
    productIds : number[],
    subCategoryIds : number[],
    printerConfigurationId?: number | null
}

File: Warehousing.Data/Entities/Role.cs
namespace Warehousing.Data.Entities
{
    public class Role : BaseClass
    {
        public int Id { get; set; }
        public string? Code { get; set; } = string.Empty;
        public string? NameEn { get; set; } = string.Empty;
        public string NameAr { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;

        public List<RolePermission> RolePermissions { get; set; }

        public List<RoleCategory> RoleCategories { get; set; } = new();
        public List<RoleProduct> RoleProducts { get; set; } = new();
        public List<RoleSubCategory> RoleSubCategories { get; set; } = new();
    }
}